### DISCUSSION TITLE
Exact reason for tunnel establishment failure

### DISCUSSION THREAD

**Discussion by nknkgithub** [strongswan/strongswan#2167] [General]
There could be multiple reasons for tunnel establishment failure such as algorithm mismatch, authentication failure,  OCSP request failure etc.
Is there any way to get the exact reason for failure other than analyzing logs?

**Comment by tobiasbrunner** [comment]
> Is there any way to get the exact reason for failure other than analyzing logs?

The [_error-notify_ plugin](https://docs.strongswan.org/docs/5.9/plugins/error-notify.html) does provide relatively specific errors.

**Reply by nknkgithub** [reply]
We are using strongswan client(strongswan-5.9.1-r0.apk) inside an alpine based container and we add the package using "apk add strongswan-5.9.1-r0.apk". Is there any way to get this error plugin without running configure and compile?

**Reply by tobiasbrunner** [reply]
No, since that package doesn't contain the plugin.

**Reply by nknkgithub** [reply]
Thanks @tobiasbrunner .
We want to use error-notify plugin to get relatively specific errors.
We are using strongswan client(strongswan-5.9.1-r0.apk) inside an alpine based container and we add the package using "apk add strongswan-5.9.1-r0.apk. 

How can we achieve this?

**Reply by tobiasbrunner** [reply]
Why would you use such an old version when [alpine ships the latest version](https://pkgs.alpinelinux.org/package/edge/main/x86_64/strongswan)? Anyway, since neither package includes that particular plugin, you have to [build from sources](https://docs.strongswan.org/docs/5.9/install/install.html#_compile_strongswan_yourself).
